roles/common/tasks/main.yml

```diff
@@ -10,7 +16,7 @@
    - unzip
    - which
    - java-1.7.0-openjdk-devel
-  when: ansible_pkg_mgr == 'yum'
+  when: ansible_pkg_mgr == 'yum' and (java_version | failed or java_version.stdout | version_compare('1.7', '<'))
```

Comment:
Yes @ketan, that's correct. @boukeversteegh, can you extract the java installation to a separate task to ensure the other basic dependencies are still installed?